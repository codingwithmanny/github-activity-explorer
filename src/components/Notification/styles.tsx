// IMPORTS
// --------------------------------------------------------
import styled from 'styled-components';
import color from 'color';
import { Colors, ScreenSizes } from '../../styles/DefaultStyles';

// TYPES
// --------------------------------------------------------
interface NotificationStylesProps {
  show: boolean;
}

// MAIN STYLES
// --------------------------------------------------------
/**
 * container styled component within <Notification />
 * @constant {JSX} styled component
 */
export const NotificationStyles = styled.div<NotificationStylesProps>`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1000;
  padding: 0 30px;

  > div {
    display: block;
    height: 40px;
    line-height: 40px;
    margin: 0 auto;
    padding: 0 20px;
    white-space: nowrap;
    background: ${Colors.red};
    color: ${color(Colors.white).alpha(0.8).toString()};
    overflow: hidden;
    text-overflow: ellipsis;
    border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;
    width: calc(100% - 160px);
    transition: all 250ms ease-in-out 0s;
    transform: ${(props) =>
      props.show ? 'translateY(0px)' : 'translateY(-40px)'};
  }

  @media (min-width: ${ScreenSizes.medium}) {
    > div {
      text-align: center;
    }
  }
`;
