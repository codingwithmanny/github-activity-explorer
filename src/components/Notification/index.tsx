// IMPORTS
// --------------------------------------------------------
import React from 'react';

// Styled Components
import { NotificationStyles } from './styles';

// Presentation Components
// TYPES
// --------------------------------------------------------
interface NotificationProps {
  message: string | null;
}

// MAIN PRESENTATION COMPONENT
// --------------------------------------------------------
/**
 * Notification
 * @param {Object} props component props
 * @returns {JSX} React compoennt
 */
const Notification: React.FC<NotificationProps> = (props) => {
  // State / Props
  const { message } = props;

  // Render
  return (
    <NotificationStyles show={(message && message.length > 0) || false}>
      <div>{message && message}</div>
    </NotificationStyles>
  );
};

// EXPORTS
// --------------------------------------------------------
export default Notification;
