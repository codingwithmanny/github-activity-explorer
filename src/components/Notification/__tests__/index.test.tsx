// IMPORTS
// --------------------------------------------------------
import React from 'react';
import renderer from 'react-test-renderer';

// TESTING
// --------------------------------------------------------
import Notification from '../';

// TEST
// --------------------------------------------------------
/**
 *
 */
test('Renders <Notification /> No Message', () => {
  const tree = renderer.create(<Notification message={null} />).toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 *
 */
test('Renders <Notification /> With Message', () => {
  const tree = renderer
    .create(<Notification message={'Something went wrong.'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
