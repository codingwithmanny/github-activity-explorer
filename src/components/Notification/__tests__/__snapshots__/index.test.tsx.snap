// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Notification /> No Message 1`] = `
<div
  className="sc-AxjAm dXYzTG"
>
  <div />
</div>
`;

exports[`Renders <Notification /> With Message 1`] = `
<div
  className="sc-AxjAm NnIan"
>
  <div>
    Something went wrong.
  </div>
</div>
`;
